Build.fsx

```diff
@@ -3,20 +3,75 @@
 open Fake
 open Fake.Testing
 open System
+open System.Diagnostics;
+open System.Text.RegularExpressions
 
 let releaseFolder = "Release"
 let nunitToolsFolder = "Packages/NUnit.Runners.2.6.2/tools"
 let nuGetOutputFolder = "NuGetPackages"
 let solutionsToBuild = !! "Src/*.sln"
 let processorArchitecture = environVar "PROCESSOR_ARCHITECTURE"
 
+type BuildVersionInfo = { assemblyVersion:string; fileVersion:string; nugetVersion:string }
+let calculateVersionFromGit buildNumber =
+    // Example of output for a release tag: v3.50.2-288-g64fd5c5b, for a prerelease tag: v3.50.2-alpha1-288-g64fd5c5b
+    let desc = Git.CommandHelper.runSimpleGitCommand "" "describe --tags --long --match=v*"
+
+    let result = Regex.Match(desc, @"^v(?<maj>\d+)\.(?<min>\d+)\.(?<rev>\d+)(?<pre>-\w+\d*)?-(?<num>\d+)-g(?<sha>[a-z0-9]+)$", RegexOptions.IgnoreCase).Groups
+    let getMatch (name:string) = result.[name].Value
+
+    let major, minor, revision, preReleaseSuffix, commitsNum, sha =
+        getMatch "maj" |> int, getMatch "min" |> int, getMatch "rev" |> int, getMatch "pre", getMatch "num" |> int, getMatch "sha"
+
+    
+    let assemblyVersion = sprintf "%d.%d.%d.0" major minor revision
+    let fileVersion = sprintf "%d.%d.%d.%d" major minor revision buildNumber
+    
```

Comment:
I'll try to trim lines to 120 characters whenever that is possible. However there are places when that would be hard (e.g. line with regular expression will look less readable if splitted into multiple lines).